File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/HeapFileRepository.java
package com.github.ol_loginov.heaplibweb.repository;

import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface HeapFileRepository {
	List<HeapFile> findAllByOrderByLoadStartDesc();

	List<HeapFile> findAllByStatusIn(List<HeapFileStatus> statuses);

	List<HeapFile> findAllByStatusNotIn(List<HeapFileStatus> statuses);

	Optional<HeapFile> findFirstByRelativePathOrderByIdDesc(String relativePath);
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/HeapFileRepository.kt
package com.github.ol_loginov.heaplibweb.repository

import com.github.ol_loginov.heaplibweb.repository.heap.HeapRepositories

interface HeapFileRepository {
    fun persist(entity: HeapFile): HeapFile
    fun merge(entity: HeapFile)

    fun findById(entityId: Int): HeapFile?

    fun findAllByOrderByLoadStartDesc(): List<HeapFile>

    fun findAllByStatusIn(statuses: List<HeapFileStatus>): List<HeapFile>

    fun findAllByStatusNotIn(statuses: List<HeapFileStatus>): List<HeapFile>

    fun findFirstByPathOrderByIdDesc(path: String): HeapFile?

    fun getHeapRepositories(heapFile: HeapFile): HeapRepositories
}